Plugins/Public/base_plugin/PlayerCommands.cpp

```diff
@@ -127,10 +127,49 @@ namespace PlayerCommands
 		pub::Player::PopUpDialog(client, caption, message, POPUPDIALOG_BUTTONS_CENTER_OK);
 	}
 
+	bool RateLimitLogins(uint client, PlayerBase *base, wstring charname)
+	{
+		uint curr_time = (uint)time(0);
+		uint big_penalty_time = 300;
+		uint amount_of_attempts_to_reach_penalty = 15;
+
+		//initiate
+		if (base->unsuccesful_logs_in_a_row.find(charname) == base->unsuccesful_logs_in_a_row.end())

```

Comment:
I wonder if we should be keying these maps by client ID rather than char name

Reply:
considering that players can leave to free client ID and another one to log in and get the same client ID, we will have the possibility of bug when a random player has a cooldown timeout.

and if we would remove cooldown with player disconnect, it would open an easy way to nullify cooldown